web/src/components/AddPluginConfigurationPage.js

```
@@ -0,0 +1,216 @@
+                    iconElementLeft={<IconButton><NavigationArrowBack /></IconButton>}
+                    onLeftIconButtonTouchTap={() => {
+                        this.props.history.goBack();
+                    }}
+                />
+                <div style={style.body}>
+                    <TextField id='configuration-name' value={this.state.configurationName}
+                               floatingLabelText='Configuration name'
+                               onChange={this._onNameChanged}/>
+                    <br/>
+                    <br/>
+                    <h3>Config fields</h3>
+                    {Object.keys(plugin.configFields).map(this._createEntryForConfigField)}
+                    <br/>
+                    <br/>
+                    <RaisedButton
+                        label="Add configuration"
+                        primary={true}
+                        onTouchTap={this._addConfigListener}
+                    />
+                </div>
+                <Snackbar
+                    open={this.state.showConfigurationAdded}
+                    message="Configuration added"
+                    autoHideDuration={3000}
+                    onRequestClose={this._onConfigurationAddedSnackbarClosed}
+                />
+                <Dialog
+                    actions={invalidInputDialogActions}
+                    modal={false}
+                    title="Error"
+                    open={this.state.invalidInputDialogOpen}
+                    onRequestClose={this._onInvalidInputsDialogClosed}>
+                    What about Name? You have to input it as well!
+                </Dialog>
+            </div>
+        );
+    }
+
+    _createEntryForConfigField = (configField) => {
+        const config = this.props.plugin.configFields[configField];
+        let entry;
+        if (config instanceof Array) {
+            entry = this._createDropDown(configField, config);
+        } else {
+            switch (config) {
```

is this a free text field? or did you mean to use [typeof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)?

also I'd fail in the `default` instead of assuming it's a text field